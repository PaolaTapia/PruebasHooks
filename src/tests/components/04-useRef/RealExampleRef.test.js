import { shallow } from 'enzyme';
import React from 'react';
import { MultipleCustomHooks } from '../../../Componentes/03-examples/MultipleCustomHooks';
import { RealExampleRef } from '../../../Componentes/04-useRef/RealExampleRef';
jest.mock('../../../Componentes/03-examples/MultipleCustomHooks');

describe('Pruebas en RealExampleRef', () => {
     const wrapper= shallow(<RealExampleRef />); 

     test('Debe de mostrarse correctamente', () => {  
     expect(wrapper).toMatchSnapshot();  
     expect(wrapper.find(MultipleCustomHooks).exists()).toBe(false); 
   }); 


     test('Debe de mostrarse el componente', () => {
          wrapper.find('button').simulate('click'); 
          expect(wrapper.find(MultipleCustomHooks).exists()).toBe(true); 
     });
     
     
})
