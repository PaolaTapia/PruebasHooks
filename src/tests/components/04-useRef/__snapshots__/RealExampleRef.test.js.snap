// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en RealExampleRef Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    RealExRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
